### PR TITLE
Mpolyquo orderings

### PR DESCRIPTION
**Pull request by ederc**
Adds keyword argument `ordering` to `quo()`.

### CODE REVIEW

**Comment by wdecker** [comment]
@ederc Thanks for doing this!
Please note that there are two more definitions for quo immediately following the other one:

function quo(R::MPolyRing, I::Vector{<:MPolyRingElem})
  return quo(R, ideal(I))
end

function quo(R::MPolyRing, f::MPolyRingElem...)
  return quo(R, ideal(collect(f)))
end

Please add ordering as well.

File: src/Rings/MPolyQuo.jl [hunk 1]
```diff
@@ -894,12 +892,12 @@ function quo(R::MPolyRing, I::MPolyIdeal)
   return q, MapFromFunc(R, q, im, pr)
 end
 
-function quo(R::MPolyRing, I::Vector{<:MPolyRingElem})
-  return quo(R, ideal(I))
+function quo(R::MPolyRing, I::Vector{<:MPolyRingElem}; ordering::MonomialOrdering = default_ordering(R))
+  return quo(R, ideal(I), ordering=ordering)
```
Comment (lgoettgens) [inline src/Rings/MPolyQuo.jl]:
```suggestion
  return quo(R, ideal(I); ordering)
```

File: src/Rings/MPolyQuo.jl [hunk 2]
```diff
@@ -894,12 +892,12 @@ function quo(R::MPolyRing, I::MPolyIdeal)
...
 end
 
-function quo(R::MPolyRing, f::MPolyRingElem...)
-  return quo(R, ideal(collect(f)))
+function quo(R::MPolyRing, f::MPolyRingElem...; ordering::MonomialOrdering = default_ordering(R))
+  return quo(R, ideal(collect(f)), ordering=ordering)
```
Comment (lgoettgens) [inline src/Rings/MPolyQuo.jl]:
```suggestion
  return quo(R, ideal(collect(f)); ordering)
```

File: test/Rings/MPolyQuo.jl [hunk 3]
```diff
@@ -1,10 +1,13 @@
 @testset "MPolyQuoRing" begin
   R, (x,y) = polynomial_ring(QQ, ["x", "y"])
-
   f = y^2+y+x^2
   C = ideal(R, [f])
-  Q, = quo(R, C)
 
+  P, = quo(R, C, ordering=lex(R))
```
Comment (lgoettgens) [inline test/Rings/MPolyQuo.jl]:
```suggestion
  P, = quo(R, C; ordering=lex(R))
```

File: test/Rings/MPolyQuo.jl [hunk 4]
```diff
@@ -18,6 +21,20 @@
 
   @test xx^2 == xx * xx
   @test xx^0 == one(Q)
+
+  J = ideal(R, [x-y^2, x^2-y^3+1])
+  A, p = quo(R, J)
+  f = A(y^2-y^4)
+  @test simplify(f) == A(-x*y + x + 1)
+  A, p = quo(R, J, ordering=lex(R))
```
Comment (lgoettgens) [inline test/Rings/MPolyQuo.jl]:
```suggestion
  A, p = quo(R, J; ordering=lex(R))
```

File: test/Rings/MPolyQuo.jl [hunk 5]
```diff
@@ -18,6 +21,20 @@
...
+  f = A(y^2-y^4)
+  @test simplify(f) == A(-y^3 + y^2 + 1)
+
+  A,p = quo(R, [x^3+y,y^2], ordering=lex(R))
```
Comment (lgoettgens) [inline test/Rings/MPolyQuo.jl]:
```suggestion
  A,p = quo(R, [x^3+y,y^2]; ordering=lex(R))
```

File: test/Rings/MPolyQuo.jl [hunk 6]
```diff
@@ -18,6 +21,20 @@
...
+  @test A.I == ideal(R, [x^3+y, y^2])
+
+  B,q = quo(R, x^3+y,y^2, ordering=lex(R))
```
Comment (lgoettgens) [inline test/Rings/MPolyQuo.jl]:
```suggestion
  B,q = quo(R, x^3+y,y^2; ordering=lex(R))
```

**Review by lgoettgens — APPROVED** [review]
Looks good to me! I have added a few minor suggestions to make it clearer there `ordering` is a kwarg.

**Review by wdecker — APPROVED** [review]
Fine with me, but why have so many tests been cancelled?

**Comment by wdecker** [comment]
@ederc Please skip outdated comment

**Comment by ederc** [comment]
> @ederc Please skip outdated comment

It is already removed since last Friday.

File: src/Rings/MPolyQuo.jl [hunk 7]
```diff
@@ -11,7 +11,8 @@
   SQRGB::Singular.sideal # Singular Groebner basis defining quotient ring
 
   function MPolyQuoRing(R::MPolyRing, I::MPolyIdeal, ordering::MonomialOrdering = default_ordering(R))
-    @assert base_ring(I) === R
+    @req base_ring(I) === R "Base ring must be the same."
```
Comment (wdecker) [inline src/Rings/MPolyQuo.jl]: Unfortunately, this change causes problems in some old tests. Can you please stick to the old code line here.

Comment (wdecker) [reply src/Rings/MPolyQuo.jl]: I mean this change:

@assert base_ring(I) === R
@req base_ring(I) === R "Base ring must be the same."

**Pull request merged**